### ISSUE TITLE
Learning tab does not support file upload out of the box

### ISSUE THREAD

**Issue by Glandos** [vstakhov/rspamd-interface#5]
Maybe I'm missing something, but from a fresh installation on Debian/unstable (official Debian package 0.8.3), if I go to the Learning tab, and try to upload a .eml file as a Spam, I have "Upload failed" and the following log messages from the daemon:

	avril 11 21:56:45 complexe rspamd[16499]: 2015-04-11 21:56:45 #16503(controller) rspamd_controller_check_password: using password as enable_password for a privileged command
	avril 11 21:56:45 complexe rspamd[16499]: 2015-04-11 21:56:45 #16503(controller) process_message: cannot construct mime from stream
	avril 11 21:56:45 complexe rspamd[16499]: 2015-04-11 21:56:45 #16503(controller) rspamd_task_process: processing of message failed
	avril 11 21:56:45 complexe rspamd[16499]: 2015-04-11 21:56:45 #16503(controller) rspamd_controller_handle_learn_common: filters cannot be processed for undef
	avril 11 21:56:51 complexe rspamd[16499]: 2015-04-11 21:56:51 #16503(controller) rspamd_controller_error_handler: http error occurred: IO read error: Bad file descriptor

Is something missing somewhere?

**Comment by Amediga** [comment]
The same problem exist in fresh installation on Centos 7 (Official Centos package 0.9.4)

Log message:

May 27 11:09:52 localhost rspamd: 2015-05-27 11:09:52 #2415(controller) process_message: construct mime parser from string length 12037
May 27 11:09:52 localhost rspamd: 2015-05-27 11:09:52 #2415(controller) process_message: cannot construct mime from stream
May 27 11:09:52 localhost rspamd: 2015-05-27 11:09:52 #2415(controller) rspamd_task_process: processing of message failed
May 27 11:09:52 localhost rspamd: 2015-05-27 11:09:52 #2415(controller) rspamd_controller_handle_learn_common: filters cannot be processed for undef
May 27 11:09:52 localhost rspamd: 2015-05-27 11:09:52 #2415(controller) rspamd_task_free: free pointer 0000000000E294D0
May 27 11:09:58 localhost rspamd: 2015-05-27 11:09:58 #2415(controller) rspamd_controller_error_handler: http error occurred: IO read error: Bad file descriptor

**Comment by vstakhov** [comment]
You need to upload the complete mime messages (meaning that they should have headers and the appropriate mime payload).

**Comment by Amediga** [comment]
.eml file contains all of headers  and mime payloads. Eml file can be opened from mail clients clearly. Also there is no mime issue or something like that. Can that be about html part of it because of that log:

rspamd_controller_error_handler: http error occurred: IO read error: Bad file descriptor

?

**Comment by leondrefoto** [comment]
I confirm. Version 1.0.4, Debian 8 Jessie.

Uploading via HTTP does not work (Upload failed). 
While uploading the same EML file via SSH to rspamd server and feeding it by rspamc then results "success = true".

**Comment by vstakhov** [comment]
Does it work if you paste it into the textarea field?

**Comment by leondrefoto** [comment]
Yes, if I paste raw content from the EML file in text area it works fine!

**Comment by leondrefoto** [comment]
There is debug while uploading the file
```
2015-10-06 20:36:12 #7209(controller) <2155b5>; csession; rspamd_controller_check_password: allow unauthorized connection from a trusted IP XX.XX.XX.XX
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: read hostname header, value: xxxx.xxxx.xx:11334
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Accept
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Accept-Language
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Accept-Encoding
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: X-Requested-With
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Cache-Control
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: wrong header: Referer
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Content-Length
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Content-Type
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_protocol_handle_headers: unknown header: Connection
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_task_load_message: got input of length 0
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_task_process: completed stage 1
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_task_process: completed stage 2
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_message_parse: construct mime parser from string length 28727
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_message_parse: cannot construct mime from stream
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_task_process: task is processed
2015-10-06 20:36:12 #7209(controller) <2155b5>; csession; rspamd_controller_handle_learn_common: <undef> message cannot be processed
2015-10-06 20:36:12 #7209(controller) <d6c69a>; task; rspamd_task_free: free pointer 0000000001A25220
2015-10-06 20:36:12 #7209(controller) <2155b5>; csession; rspamd_controller_finish_handler: destroy session 00000000019AE8A0
```
